tools/kwiver_tool_runner.cxx

```diff
@@ -0,0 +1,255 @@
+/*ckwg +29
+ * Copyright 2018 by Kitware, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#include "kwiver_applet.h"
+#include "applet_context.h"
+
+#include <vital/plugin_loader/plugin_manager.h>
+#include <vital/plugin_loader/plugin_factory.h>
+#include <vital/exceptions/base.h>
+
+#include <kwiversys/SystemTools.hxx>
+
+#include <cstdlib>
+#include <iostream>
+#include <fstream>
+#include <exception>
+#include <memory>
+
+
+typedef kwiver::vital::implementation_factory_by_name< kwiver::tools::kwiver_applet > applet_factory;
+typedef std::shared_ptr< kwiver::tools::applet_context > applet_context_t;
+
+
+// ============================================================================
+class command_line_parser
+{
+public:
+  command_line_parser( int p_argc, char *** p_argv )
+    : m_argc(0)
+    , m_argv(0)
+    , m_state(0)
```

Comment:
Is there a reason that this needs to be a member variable rather than a local variable?